images/gitlab-delete-stale-branches/delete_stale_branches.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+import json
+import os
+import re
+import urllib.parse
+from datetime import datetime, timedelta, timezone
+
+import requests
+import sentry_sdk
+
+sentry_sdk.init(
+    # This cron job only runs once a week,
+    # so just record all transactions.
+    traces_sample_rate=1.0,
+)
+
+
+GITLAB_API_URL = "https://gitlab.spack.io/api/v4/projects/2"
+AUTH_HEADER = {
+    "PRIVATE-TOKEN": os.environ.get("GITLAB_TOKEN", None)
+}
+
+
+def paginate(query_url):
+    """Helper method to get all pages of paginated query results"""
+    results = []
+
+    while query_url:
+        resp = requests.get(query_url, headers=AUTH_HEADER)
+
+        if resp.status_code == 401:
+            print(" !!! Unauthorized to make request, check GITLAB_TOKEN !!!")
+            resp.raise_for_status()
```

Comment:
Should we just call `raise_for_status` here unconditionally? Any `4xx` or `5xx` response would be invalid and not processable.  

```suggestion
        resp = requests.get(query_url, headers=AUTH_HEADER)
        if resp.status_code == 401:
            print(" !!! Unauthorized to make request, check GITLAB_TOKEN !!!")

        resp.raise_for_status()

```